modules/querying/pages/data-types.adoc

```diff
@@ -451,18 +451,42 @@ include::appendix:example$investment_net/datatypes_udt_results.json[]
 
 == `FILE` Object
 
-A `FILE` object is a sequential data storage object, associated with a text file on the local machine.
+A `FILE` object is a sequential data storage object, associated with a text file on the local machine or on S3 bucket.
 
 [NOTE]
 ====
 When referring to a `FILE` object, we always capitalize the word `FILE` to distinguish it from ordinary files.
 ====
 
+=== Local disk file
 When a `FILE` object is declared, associated with a particular text file, any existing content in the text file will be erased.
 During the execution of the query, content written to the `FILE` will be appended to the `FILE`.
 When the query where the `FILE` was declared finishes running, the `FILE` contents are saved to the text file.
 
-A `FILE` object can be passed as a parameter to another query. When a query receives a `FILE` object as a parameter, it can append data to that `FILE`, as can every other query which receives this `FILE` object as a parameter.
+=== S3 object
+==== Define s3 file object
+The path should start with `s3://`, followed by the bucket name and the S3 path, e.g., `s3://bucket-name/queryoutput/output.csv`. During the execution of the query, content will be uploaded to the S3 bucket. Note that the S3 object cannot be modified or appended, if an S3 object with the same path already exists, it will be overwritten.
+
+==== Set S3 connection credentials
+To use an S3 path, ensure that the necessary permissions and configurations are properly set up to allow read/write access to the specified S3 bucket. Configure the S3 connection credentials using `gadmin config`:
+[source,bash]
+----
+gadmin config set GPE.QueryOutputS3AWSAccessKeyID YOUR_AWS_ACCESS_KEY_ID
+gadmin config set GPE.QueryOutputS3AWSSecretAccessKey YOUR_AWS_SECRET_ACCESS_KEY
+gadmin config apply -y
+gadmin restart gpe -y
+----
+
+==== Output
+Since S3 is a shared storage system, multiple nodes in a cluster can upload to the same S3 bucket. To handle potential conflicts and ensure unique output files, the S3 path can include a suffix based on the instance name, such as `\_GPE_{PartitionId}_{ReplicaId}`. For distributed queries, additional suffixes will be used to differentiate between the manager and worker roles on the same GPE. Specifically, suffixes like `_coordinator` and `_worker` will be added, where `_coordinator` refers to the worker manager and `_worker` refers to the worker node.
+
+==== Error code
+For S3 bucket connection errors, refer to error code `GSQL-5301`.
+
+[NOTE]
+====
+A FILE object can be passed as a parameter to another query. When a query receives a FILE object as a parameter, for a file on the local machine, it can append data to that FILE, as can every other query which receives this FILE object as a parameter. However, for an S3 path, the FILE cannot be modified or appended, it will overwrite any existing object at the same path when the query finishes running.
```

Comment:
```suggestion
A `FILE` object can be passed as a parameter to another query.
When a query receives a `FILE` object as a parameter, for a file on the local machine, it can append data to that `FILE`, as can every other query which receives this FILE object as a parameter.
However, an S3 bucket `FILE` object cannot be appended to.
When you write to an S3 path, any existing object will be overwritten.
```